### PR TITLE
[Finder] Merge regexps for performance

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

Follow up of #15826. May be superseded by an other PR from @stof, still submitting for history at least.

### CODE REVIEW

File: src/Symfony/Component/Finder/Expression/Regex.php [hunk 1]
```diff
@@ -61,7 +61,7 @@ class Regex implements ValueInterface
      */
     public static function create($expr)
     {
-        if (preg_match('/^(.{3,}?)([imsxuADU]*)$/', $expr, $m)) {
+        if (preg_match('/^(.{2,}?)([imsxuADU]*)$/', $expr, $m)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Finder/Expression/Regex.php]: Empty regexps (`//`) *are* valid

File: src/Symfony/Component/Finder/Iterator/MultiplePcreFilterIterator.php [hunk 2]
```diff
@@ -65,4 +59,28 @@ protected function isRegex($str)
...
+        foreach ($patterns as $pattern) {
+            $regex = Regex::create($this->toRegex($pattern))->render();
+
+            if (preg_match('/\W([imsxUXJ]*)$/', $regex, $match) && !preg_match('/\\\\[1-9]|\(\?P/', $regex)) {
+                $rxs[] = '(?'.$match[1].':'.substr($regex, 1, -strlen($match[0])).')';
+            } else {
+                $regexps[] = $regex;
+            }
+        }
+        if (1 < count($rxs)) {
```
Comment (stloyd) [inline src/Symfony/Component/Finder/Iterator/MultiplePcreFilterIterator.php]: `!empty()`?

Comment (nicolas-grekas) [inline src/Symfony/Component/Finder/Iterator/MultiplePcreFilterIterator.php]: `!empty()` would mean `0 < count($rxs)`

Comment (stloyd) [inline src/Symfony/Component/Finder/Iterator/MultiplePcreFilterIterator.php]: Right, my bad.

**Comment by fabpot** [comment]
What do we do with this PR?

**Comment by fabpot** [comment]
Closing as this PR uses the Regex class that does not exist anymore. @stof if you have a better idea, feel free to open a PR :)

**Pull request closed without merging**